README.md

```diff
@@ -0,0 +1,51 @@
+[![CharmHub Badge](https://charmhub.io/github-runner-operator/badge.svg)](https://charmhub.io/github-runner-operator)
+[![Promote charm](https://github.com/canonical/github-runner-operator/actions/workflows/promote_charm.yaml/badge.svg)](https://github.com/canonical/github-runner-operator/actions/workflows/promote_charm.yaml)
+[![Discourse Status](https://img.shields.io/discourse/status?server=https%3A%2F%2Fdiscourse.charmhub.io&style=flat&label=CharmHub%20Discourse)](https://discourse.charmhub.io)
+
+# GitHub runner
+
+## Description
+
+This machine charm creates self-hosted GitHub runners. Each unit of this charm will start a configurable number of LXD based containers and virtual
+machines to host them. Each runner performs only one job, after which it unregisters from GitHub to ensure that each job runs in
+a clean environment.
+
+The charm will periodically check the number of idle runners and spawn or destroy runners as necessary to match the number provided by configuration of
+runners. Both the reconciliation interval and the number of runners to maintain are configurable.
+
+## Usage
+
+There are two mandatory configuration options - `path` and `token`.
+* `path` determines the organization or repository that the runner will be registered with;
+* `token` is a [GitHub Personal Access Token (PAT)](https://github.com/settings/tokens) (note: this is not the same as the token given in the Add a Runner instructions). The PAT token requires either:
+  * the **`repo`** ("Full control of private repositories") permission for
+use with repositories or;
+  * both the **`repo`** and **`admin:org`** ("Full control of orgs and teams, read and write org projects") permissions for use with an organization. This is necessary because the charm will create and remove runners as needed to ensure that each runner executes only one job to protect jobs from leaking information to other jobs running on the same runner.
```

Comment:
The fine-grained token are in beta right now, but it might be good to include the instruction for fine-graned tokens.
https://docs.github.com/en/rest/overview/permissions-required-for-fine-grained-personal-access-tokens?apiVersion=2022-11-28

Reply:
Minor comments